server/plrhand.cpp

```diff
@@ -792,17 +806,22 @@ void handle_diplomacy_cancel_pact(struct player *pplayer,
   new_type = cancel_pact_result(old_type);
 
   // avoid accidentally breaking alliances
-  if (new_type == DS_WAR) {
+  if (protect_alliances && new_type == DS_WAR) {
     bool blocking_alliances = false;
 
     players_iterate_alive(pplayer3)
     {
       if (pplayer3 != pplayer && pplayer3 != pplayer2
           && pplayers_allied(pplayer3, pplayer)
           && pplayers_allied(pplayer3, pplayer2)) {
-        blocking_alliances = true;
-
         if (players_on_same_team(pplayer, pplayer3)) {
+          // if pplayer3 is an AI, let them break their alliance as we
+          // can't coordinate with them in a way, that they would cancel
+          // alliances in preparation of war on team level.
```

Comment:
One more argument for diplomacy to be handled at the team level :sweat_smile: (and by the human when there's one on the team)

Reply:
It would definitely make things clearer, but it would also be a lot of work 😅